lib/Parser/Parse.cpp

```diff
@@ -1398,15 +1398,52 @@ ParseNodePtr Parser::AddVarDeclNode(IdentPtr pid, ParseNodePtr pnodeFnc)
     return pnode;
 }
 
-ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr pid)
+Js::PropertyId Parser::EnsurePropertyId(IdentPtr pid)
 {
-    ParseNodePtr declNode = CreateBlockScopedDeclNode(pid, knopConstDecl);
+    Js::PropertyId propertyId = pid->GetPropertyId();
+    if (propertyId == Js::Constants::NoProperty)
+    {
+        propertyId = m_scriptContext->GetOrAddPropertyIdTracked(pid->Psz(), pid->Cch());
+        pid->SetPropertyId(propertyId);
+    }
+    return propertyId;
+}
 
-    declNode->sxVar.sym->SetIsModuleExportStorage(true);
+ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr localName)
+{
+    ParseNodePtr declNode = CreateBlockScopedDeclNode(localName, knopConstDecl);
+    Symbol* sym = declNode->sxVar.sym;
+    
+    sym->SetIsModuleExportStorage(true);
 
     return declNode;
 }
 
+Symbol* Parser::MarkIdentifierReferenceIsModuleExport(IdentPtr localName)
+{
+    PidRefStack* pidRef = this->PushPidRef(localName);
```

Comment:
Well, a statement at global scope can't export a local name. In your example, we're exporting a global name. I think the solution we discussed originally (or maybe I only dreamed that we discussed it) was this: At the time we call PushPidRef for the identifier reference in the export statement, mark a bit on the pidref to indicate that the symbol it binds to is exported. Then, in BindPidRefsInScopeImpl, if the ref has the "exported" bit set, mark the symbol as exported. That should eliminate the search. At that point, we should be able to assert that the symbol is global.